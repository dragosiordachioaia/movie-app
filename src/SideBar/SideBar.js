import React, { Component } from "react";

import "./SideBar.scss";

import RatingFilter from "RatingFilter/RatingFilter";
import Genres from "Genres/Genres";

export default class SideBar extends Component {
  render() {
    return (
      <div className="side-bar">
        <div className="inner-container">
          <RatingFilter
            minRating={this.props.minRating}
            onChange={this.props.onMinRatingChange}
          />
          <Genres
            genres={this.props.genres}
            onGenreCheck={this.props.onGenreCheck}
          />
        </div>
        <button className="expand" onClick={this.props.onExpand}>
          <i
            className={`fa fa-chevron-${this.props.expanded ? "up" : "down"}`}
          />
        </button>
      </div>
    );
  }
}
