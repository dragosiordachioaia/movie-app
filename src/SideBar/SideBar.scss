@import "../variables";

.side-bar {
  box-sizing: border-box;
  background-color: $color-primary-alt;
  color: #eee;
  padding-left: 1em;
  padding-right: 1em;
  overflow: hidden;
  height: $sidebar-height;
  width: 100%;
  position: relative;
  transition: height 500ms;
  transition-timing-function: ease-in-out;
  -webkit-overflow-scrolling: touch;

  @media (min-width: $SM) {
    height: 100%;
    width: $sidebar-width;
  }

  .inner-container {
    height: calc(100% - 2em);
    overflow: auto;
  }

  p {
    font-weight: 700;
    font-size: 0.9em;
  }

  button.expand {
    position: absolute;
    bottom: 0em;
    color: white;
    background-color: transparent;
    border: 0;
    transform: translateX(-50%);
    left: 50%;
    outline: 0;
    padding: 0.7em;

    @media (min-width: $SM) {
      display: none;
    }

    .fa {
      font-size: 1.2rem;
      position: relative;
      top: -0.1em;
    }
  }
}
