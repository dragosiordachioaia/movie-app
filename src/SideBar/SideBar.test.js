import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SideBar from "./SideBar";

jest.mock("RatingFilter/RatingFilter", () => "RatingFilter");
jest.mock("Genres/Genres", () => "Genres");

let component;
let instance;

beforeEach(() => {
  component = renderer.create(<SideBar />);
  instance = component.getInstance();
});

it("renders correctly when not expanded", () => {
  instance.props = { expanded: false };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
it("renders correctly when expanded", () => {
  instance.props = { expanded: true };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
