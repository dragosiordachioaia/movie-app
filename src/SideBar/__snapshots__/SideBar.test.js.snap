// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when expanded 1`] = `
<div
  className="side-bar"
>
  <div
    className="inner-container"
  >
    <RatingFilter />
    <Genres />
  </div>
  <button
    className="expand"
  >
    <i
      className="fa fa-chevron-up"
    />
  </button>
</div>
`;

exports[`renders correctly when not expanded 1`] = `
<div
  className="side-bar"
>
  <div
    className="inner-container"
  >
    <RatingFilter />
    <Genres />
  </div>
  <button
    className="expand"
  >
    <i
      className="fa fa-chevron-down"
    />
  </button>
</div>
`;
